### ISSUE TITLE
Cannot receive a deferred message using the go client, while it's possible receive it using the nodejs client  

### ISSUE THREAD

**Issue by 3mard** [Azure/azure-service-bus-go#241]
Hi 
Thanks for your hard work!

We are facing an issue in which we cannot receive the deferred message using the go client
We get the following error : 
`server error link 5e7768a7-4b8c-4516-8c0c-e3214e50c273: status code 500 and description: Messages not found. One or more sequence numbers missing. TrackingId:abfb3c60-ff65-4146-bc1e-426c52aa51ce_B46, SystemTracker:uat-onservicebus:Topic:<topic_name>|<sub_name>|$deadletterqueue, Timestamp:2021-08-04T16:00`

However, using the Nodejs client we are able to receive it 

This is how we are trying to receive the message using the go client 

```
	err = dlq.ReceiveDeferredWithMode(ctx, servicebus.HandlerFunc(func(c context.Context, m *servicebus.Message) error {
		fmt.Printf("%s", m.Data)
		m.Abandon(c)
		return nil
	}), servicebus.PeekLockMode, 183)

	if err != nil {
		log.Fatal(err)
	}
```
We are sure that 183 actually exists in our queue, and it is in fact a deferred message
The error message that we are getting is the same error message for a sequence number that doesn't exist in the queue 
Our guess is that somehow the go client is not transmitting the sequence number correctly 

Thanks in advance.

**Comment by richardpark-msft** [comment]
Hi @3mard, I'll take a look at that.

**Comment by richardpark-msft** [comment]
I'll update and release once I this PR is merged in go-amqp:  https://github.com/Azure/go-amqp/pull/43

Details for others (and for @jhendrixMSFT):

The bug was actually in the way that go-amqp was encoding arrayInt64. go-amqp has a nice facility where it will downsize types if they can fit within the range of a smaller type, which can save on bytes going over the wire. The sequence numbers we pass in for this are int64's, but if it can it'll try to use a int8 to encode them.

The go-amqp code was testing the range, but using math.Uint8 (too high!). That will be fixed when the the go-amqp PR is merged https://github.com/Azure/go-amqp/pull/43.

Our automated tests, at the Service Bus level, don't catch this because our sequence numbers on a brand new queue (ie: created at test time) are going to be naturally low.

**Comment by richardpark-msft** [comment]
Hi @3mard,

This fix has now been released as part of https://github.com/Azure/azure-service-bus-go/releases/tag/v0.10.15.

**Comment by 3mard** [comment]
Thanks @richardpark-msft 

I just tested it ,it works
